helix-term/src/commands/lsp.rs

```diff
@@ -154,7 +154,8 @@ impl ui::menu::Item for PickerDiagnostic {
         Spans::from(vec![
             Span::raw(path),
             Span::styled(&self.diag.message, style),
-            Span::styled(code, style),
+            Span::styled(" ", style),
+            Span::styled(format!("({code})"), style),
```

Comment:
This will draw the extra space and empty parens if the diagnostic's `code` is `None`. Instead the formatting can be done at the `let code` part:

```rust
let code = self
    .diag
    .code
    .as_ref()
    .map(|c| {
        format!(
            " ({})",
            match c {
                NumberOrString::Number(n) => n.to_string(),
                NumberOrString::String(s) => s.clone(),
            }
        )
    })
    .unwrap_or_default();
```

Reply:
```rust
let code = match self.diag.code.as_ref() {
    Some(NumberOrString::Number(n)) => format!(" ({n})"),
    Some(NumberOrString::String(s)) => format!(" ({s})"),
    None => String::new(),
};
```

Will allocate less (no need for `clone()` nor `to_string()`) and be shorter